integration_tests/tests/assert_fetch_objects_equal.sql

```diff
@@ -0,0 +1,19 @@
+-- depends_on: {{ ref('data_fetch') }}
+
+{% set expected_dictionary={
+    'col_1': (1, 2, 3),
+    'col_2': ('a', 'b', 'c'),
+    'col_3': (4.0, 5.0, none)
+} %}
+
+{% set actual_dictionary=dbt_utils.fetch(
+    "select * from {{ ref('data_fetch') }}"
+) %}
+
+{% if actual_dictionary | lower == expected_dictionary | lower %}
```

Comment:
I'm not sure if you can call `lower` on a dictionary - I'd recommend plucking out the fields that you care about, so:
```
{% if actual_dictionary['col_1'] | lower != expected_dictionary['col_1'] | lower %}
  select 1
{% elif actual_dictionary['col_2'] | lower != expected_dictionary['col_2'] | lower %}
  select 1
{% elif ... %}

{% else %}
  select 1 limit 0
{% endif %}
```

The `limit 0` approach tends to work better than `where false`, as I think there's a database out there (postgres? bigquery? I forget) which doesn't let you use a `where` predicate if you don't also have a `from` clause in the query.

Overall though, this is a slick looking test!

Reply:
This is actually something I'm having troubles/need help with. The fetch macro returns numbers in the form ```decimal('1'), decimal('2'), decimal('3')``` rather than ```1,2,3``` . I'm not sure how this would (or should) be converted.

Reply:
Has this decimal('1') been fixed? I'm having similar issues with dates